### PR TITLE
enh: ability to continue unclaimed conversations 

### LINKED ISSUE #1195: Continue a conversation (if not yet assigned)
See https://dust4ai.slack.com/archives/C050A0S2Z7F/p1693381649745159

### PR DESCRIPTION
**Pull request by spolu**
Fix dust-tt/tasks#21 

When a conversation is created by API it is not assigned to a user. When that's the case we allow visiting users to continue the conversation (effectively giving them ownership of the unclaimed conversation).

- The PR also fixes a few types inaccuracies.
- It also updates the Slack bot message from "View conversation" to "Continue conversation"

Screenshot from 2023-08-31 12-10-32

### CODE REVIEW

**Comment by spolu** [comment]
I capitalized "Continue Conversation" out of consistency with the "New Conversation" button cc @Duncid

**Comment by Duncid** [comment]
I would do more something like that:
image

**Comment by spolu** [comment]
So center XL ?

**Comment by spolu** [comment]
No looks like same size. Will center it

File: front/pages/api/w/[wId]/use/chats/[cId]/take.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            message: "The chat session was not found.",
+          },
+        });
+      }
+
+      if (
+        chatSession.userId !== null &&
+        chatSession.userId !== auth.user()?.id
+      ) {
+        return apiError(req, res, {
```
Comment (lasryaric) [inline front/pages/api/w/[wId]/use/chats/[cId]/take.ts:83]: This case is already handle in `takeOwnerShipOfChatSession()`:
```
if (chatSession.userId !== null && chatSession.userId !== user.id) {
    throw new Error("Chat session already belongs to another user.");
  }
```

Not a big deal but it's not needed here I think,.

**Review by lasryaric — APPROVED** [review]

**Pull request merged**